spec/jobs/get_usps_proofing_results_job_spec.rb

```diff
@@ -1,1900 +1,2549 @@
+        before do
+          subject.perform(current_time)
+        end
 
-          it_behaves_like(
-            'enrollment_with_a_status_update',
-            passed: false,
-            email_type: 'deadline passed',
-            enrollment_status: InPersonEnrollment::STATUS_EXPIRED,
-            response_json: UspsInPersonProofing::Mock::Fixtures.
-            request_expired_enhanced_ipp_results_response,
-            enhanced_ipp_enrollment: true,
+        it 'logs the job started analytic' do
+          expect(analytics).to have_received(:idv_in_person_usps_proofing_results_job_started).with(
+            enrollments_count: 0,
+            reprocess_delay_minutes: 5,
+            job_name: described_class.name,
           )
+        end
 
-          it 'logs that the enrollment expired' do
-            job.perform(Time.zone.now)
-
-            expect(pending_enrollment.proofed_at).to eq(nil)
-            expect(job_analytics).to have_logged_event(
-              'GetUspsProofingResultsJob: Enrollment status updated',
-              hash_including(
-                reason: 'Enrollment has expired',
-                job_name: 'GetUspsProofingResultsJob',
-                enhanced_ipp: true,
-              ),
-            )
-          end
-
-          it 'deactivates the associated profile' do
-            expect(pending_enrollment.profile.deactivation_reason).to be_nil
-            job.perform(Time.zone.now)
-            pending_enrollment.reload
-
-            expect(pending_enrollment.profile.active).to be false
-            expect(pending_enrollment.profile.in_person_verification_pending_at).to be_nil
-            expect(pending_enrollment.profile.deactivation_reason).to eq('verification_cancelled')
-          end
-
-          context 'when the in_person_stop_expiring_enrollments flag is true' do
-            before do
-              allow(IdentityConfig.store).to(
-                receive(:in_person_stop_expiring_enrollments).and_return(true),
-              )
-            end
-
-            it 'treats the enrollment as incomplete' do
-              job.perform(Time.zone.now)
-
-              expect(pending_enrollment.status).to eq(InPersonEnrollment::STATUS_PENDING)
-              expect(job_analytics).to have_logged_event(
-                'GetUspsProofingResultsJob: Enrollment incomplete',
-                hash_including(
-                  response_message: 'More than 7 days have passed since opt-in to IPP',
-                  job_name: 'GetUspsProofingResultsJob',
-                ),
-              )
-            end
-          end
+        it 'logs the job completed analytic' do
+          expect(analytics).to have_received(
+            :idv_in_person_usps_proofing_results_job_completed,
+          ).with(
+            **default_job_completion_analytics,
+          )
         end
       end
     end
 
-    describe 'IPP disabled' do
+    context 'when the job is not configured to run' do
```

Comment:
Nit: Can this be moved to the top of the file? I think it will make the test results easier to read because of all of the nesting when the job is enabled will be below it.

Reply:
Moved!